apps/nextra/components/landing/components/MoveCodeBlock.tsx

```diff
@@ -0,0 +1,53 @@
+import { CopyToClipboard } from "nextra/components";
+import { useRef } from "react";
+import { cn } from "utils/cn";
+import { IconMoveLang } from "./Icons";
+
+// This component is loosely based on Nextra's `Pre` component.
+// I originally tried to use `Pre` directly, but ended up needing more control
+// Reference: https://github.com/shuding/nextra/blob/main/packages/nextra/src/components/pre.tsx
+
+export interface MoveCodeBlockProps {
+  codeSnippet: TrustedHTML;
+  fileName?: string;
+  className?: string;
+}
+
+export function MoveCodeBlock({
```

Comment:
Nit: Any chance we could make this more generic to support other languages? I think I may use this for a remote codeblock component.

Maybe it could take in an `Icon` as props?